frontend/src/components/G6Graph.svelte

```diff
@@ -3,42 +3,62 @@
   import G6, { Graph, IG6GraphEvent, Item, Node } from "@antv/g6";
   import type { TimeBarData } from "../uitypes";
   import { createEventDispatcher } from "svelte";
+  import { interactiveMode} from '../store';
 
   const dispatch = createEventDispatcher();
   const graph_translation: number = 50;
 
   export let options = {};
   export let data = {};
 
-  let container: HTMLElement;
-  let graph: Graph | null;
-  let timeBarData: TimeBarData[] = [];
+    let nodePoint:number = 0;
 
-  export const reset = () => {
-    graph?.changeData({});
-    timeBarData = [];
-    graph?.render();
-    dispatch("nodeselected", null);
-  };
+    let container: HTMLElement;
+    let graph: Graph | null;
+    let timeBarData: TimeBarData[] = [];
+    export const reset = () => {
+      graph?.changeData({});
+      timeBarData = [];
+      graph?.render();
+      dispatch("nodeselected", null);
+    };
+      // This is a hack to get the graph to render the entire window width
+      export const resizeGraph = () => {
+          if (graph && container) {
+              const width = Number(
+                  window.innerWidth
+              );
+              const height = Number(
+                  window.innerHeight
+              );
+              graph.changeSize(width, height);
+          }
+      };
+    export const focusNode = (id: any) => {
+      graph.focusItem(id);
+    };
 
-    // This is a hack to get the graph to render the entire window width
-  export const resizeGraph = () => {
-        if (graph && container) {
-            const width = Number(
-                window.innerWidth
-            );
-            const height = Number(
-                window.innerHeight
-            );
-            graph.changeSize(width, height);
+     //non-InteractiveMode function to calculate how much translation the graph should make
+    export const nonInteractiveMode = (e:any , modeDisabled:boolean) =>{
+        if(modeDisabled == true){
+            graph.translate(0,0); // to disable nonInteractive mode
         }
-  };
+        else{
+            let oldNodePoint:number = nodePoint;
+            let scrollDistance:number  = 0;
+            let scrollDistanceWithRatio:number = 0;
+            let zoomRatio:number = graph.getZoom(); 
+            nodePoint = e.x - container.scrollWidth;
+            if(nodePoint != oldNodePoint){ // find out if node has moved to new position
+              scrollDistance = nodePoint - oldNodePoint; 
+              scrollDistanceWithRatio = scrollDistance*zoomRatio // to recalculate translation based on zoom ratio
+              graph.translate(-scrollDistanceWithRatio,0);
+            }
+        }
+    }
 
```

Comment:
This is all nice as well but I would add more comments, it makes it easier to read!

Reply:
Understood, will do!